language-support/hs/bindings/src/DA/Ledger/Convert.hs

```diff
@@ -303,6 +381,18 @@ raiseList f v = loop (Vector.toList v)
               [] -> return []
               x:xs -> do y <- f x; ys <- loop xs; return $ y:ys
 
+raiseMap :: forall a b c. Ord a =>
```

Comment:
This might be clearer if you generalize this a bit:
```haskell
raiseMap
    :: forall f k k' v v'. (Monad f, Ord k')
    => (k -> f k')
    -> (v -> f v')
    -> Map k v
    -> f (Map k' v')
raiseMap raiseK raiseV = fmap Map.fromList . mapM (uncurry raiseKV) . Map.toList
    where raiseKV :: k -> v -> f (k', v')
          raiseKV kLow vLow = (,) <$> raiseK kLow <*> raiseV vLow
```
You can move the handling of the `Maybe` part into `raiseV`.

Reply:
The `Maybe` is important for this abstraction, because it corresponds to the type attributed by `grpc-haskell`  to protobuf Maps. And so the caller line is simply this: 
```
events <- raiseMap raiseEventId raiseTreeEvent transactionTreeEventsById
```

Keeping the `Maybe` means generalisaion to `Monad f` is not directly possible, because the code as written includes `missing "mapElem"`, and `missing` is specific to the `Perhaps` monad.

But in any case I don't think generalising to an arbitrary monad here would be particularly helpful to the reader. This whole file is about `lower`ing and `raise`ing types, and all the `raise` functions operate in the `Perhaps` monad.

Reply:
I see, you could still generalize while keeping `Maybe` if you let the caller pass in a handler for the `Nothing` case but that’s probably not worth it. Would be nice to capture the fact that `Maybe` comes from `grpc-haskell` in a comment. That said, I would at least generalize the key type. If I see a type that is not polymorphic, I expect that there is a reason for it but here you aren’t making use of the fact that it is `Text`.

Reply:
I will generalize from `Text` to any key type. And add a comment about the `Maybe` :)